File: easy/11. Class Photos/class_photos.go
package main

import "sort"

func ClassPhotos(redShirtHeights []int, blueShirtHeights []int) bool {
	sort.Sort(sort.Reverse(sort.IntSlice(redShirtHeights)))
	sort.Sort(sort.Reverse(sort.IntSlice(blueShirtHeights)))
	var redShirtsInFirstRow bool

	if blueShirtHeights[0] > redShirtHeights[0] {
		redShirtsInFirstRow = true
	} else {
		redShirtsInFirstRow = false
	}

	for i := 0; i < len(redShirtHeights); i++ {
		if redShirtsInFirstRow {
			if redShirtHeights[i] >= blueShirtHeights[i] {
				return false
			}
		} else {
			if blueShirtHeights[i] >= redShirtHeights[i] {
				return false
			}
		}
	}

	return true
}

File: easy/11. Class Photos/class_photos.cpp
#include <algorithm>
#include <bits/stdc++.h>
#include <vector>

bool classPhotos(std::vector<int> redShirtHeights,
                 std::vector<int> blueShirtHeights) {
  std::sort(redShirtHeights.begin(), redShirtHeights.end(),
            std::greater<int>());
  std::sort(blueShirtHeights.begin(), blueShirtHeights.end(),
            std::greater<int>());

  bool colorsInFirstRow = redShirtHeights[0] < blueShirtHeights[0];
  for (int i = 0; i < redShirtHeights.size(); i++) {
    int red_height = redShirtHeights[i];
    int blue_height = blueShirtHeights[i];
    if (colorsInFirstRow) {
      if (red_height >= blue_height) {
        return false;
      }
    } else {
      if (blue_height >= red_height) {
        return false;
      }
    }
  }
  return true;
}
